File: Assets/Scripts/LuaScripts/UILogic/Loading/LoadingView.lua
-- ******************************************************************
--@file         LoadingView.lua
--@brief        加载界面的view
--@author       yufulao, yufulao@qq.com
--@createTime   2024/5/18 1:26
-- ******************************************************************
---@brief
---@class LoadingView : UIBaseView
LoadingView = Class("LoadingView", UIBaseView)

---@brief 初始化
---@public
function LoadingView:OnInit(...)
    self.animator = self.transform:GetComponent(typeof(CS.UnityEngine.Animator))
    self.mainCanvasGroup = self.transform:Find("Main"):GetComponent(typeof(CS.UnityEngine.CanvasGroup))
end

---@brief 打开窗口
---@public
function LoadingView:OpenRoot(...)
    self.mainCanvasGroup.alpha = 0;
    self.gameObject:SetActive(true);
    GameManager:StartCoroutine(Utils.PlayAnimation(self.animator, "LoadingEnter"));
end

---@brief 关闭窗口
---@public
function LoadingView:CloseRoot()
    Coroutine.start(function() self:CloseRootIEnumerator() end)
end

---@brief 关闭窗口的协程
---@private
function LoadingView:CloseRootIEnumerator()
    self.mainCanvasGroup.alpha = 1
    coroutine.yield(Utils.PlayAnimation(self.animator, "LoadingExit"))
    self.gameObject:SetActive(false)
end



return LoadingView

File: Assets/Scripts/UI/Windows/Loading/LoadingView.cs
// ******************************************************************
//@file         LoadingView.cs
//@brief        加载界面的view
//@author       yufulao, yufulao@qq.com
//@createTime   2024.05.18 01:36:05
// ******************************************************************

using System.Collections;
using UnityEngine;
using Yu;

public class LoadingView : MonoBehaviour
{
    public Animator animator;
    public CanvasGroup mainCanvasGroup;


    /// <summary>
    /// 打开窗口
    /// </summary>
    public void OpenRoot()
    {
        GameManager.Instance.StartCoroutine(OpenRootCo());
    }

    /// <summary>
    /// 关闭窗口
    /// </summary>
    public void CloseRoot()
    {
        GameManager.Instance.StartCoroutine(CloseRootCo());
    }
    
    /// <summary>
    /// 打开窗口的协程，带动画播放时长
    /// </summary>
    public IEnumerator OpenRootCo()
    {
        mainCanvasGroup.alpha = 0;
        gameObject.SetActive(true);
        yield return Utils.PlayAnimation(animator, "LoadingEnter");
    }

    /// <summary>
    /// 关闭窗口的协程，带动画播放时长
    /// </summary>
    public IEnumerator CloseRootCo()
    {
        mainCanvasGroup.alpha = 1;
        yield return Utils.PlayAnimation(animator, "LoadingExit");
        gameObject.SetActive(false);
    }
}